### PR TITLE
- Architecture components version bump & gradle api update

### PR DESCRIPTION
**Pull request by kvithayathil**
- Architecture components version bump to release
- Updating dependency configurations to the new gradle api: https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html#new_configurations

### CODE REVIEW

**Review by Tofuus — APPROVED** [review]
Keith is unavailable for the week so we'll probably have to wait until next week to get these merged in. Otherwise both PRs look fine. Good work!

**Comment by TheHollidayInn** [comment]
Thanks fixing Travis! Can you update the conflicts in both PRs?

**Comment by kvithayathil** [comment]
Just removed those conflicts

**Pull request merged**
